src/cpp/core/include/core/r_util/RActiveSessions.hpp

```diff
@@ -66,37 +85,59 @@ class ActiveSession : boost::noncopyable
 
    std::string project() const
    {
+      std::string project;
       if (!empty())
-         return readProperty("project");
-      else
-         return std::string();
+      {
+         Error error = storage_->readProperty(id_, kProject, &project);
+         if (error)
+            LOG_ERROR(error);
```

Comment:
To avoid duplication, I would put this implementation into `ActiveSession::readProperty` and leave these methods the same.

Reply:
The same for writing properties

Reply:
Reading and writing properties encapsulated to avoid repeated error handling and pivoting